modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -33,6 +33,43 @@ CoreMeshGenerator::validParams()
       "pattern",
       "A double-indexed array starting with the upper-left corner where the index"
       "represents the layout of input assemblies in the core lattice.");
+  params.addParam<bool>(
+      "mesh_periphery", false, "Determines if the core periphery should be meshed.");
+  MooseEnum periphery_mesher("PRMG PTMG", "PRMG");
+  params.addParam<MooseEnum>("periphery_generator",
+                             periphery_mesher,
+                             "The meshgenerator to use when meshing the core boundary.");
+
+  // Periphery meshing interface
+  params.addRangeCheckedParam<Real>(
+      "outer_circle_radius", "outer_circle_radius>0", "Radius of outer circle boundary.");
+  params.addRangeCheckedParam<unsigned int>(
+      "outer_circle_num_segments",
+      "outer_circle_num_segments>0",
+      "Number of radial segments to subdivide outer circle boundary.");
+  params.addRangeCheckedParam<unsigned int>(
+      "periphery_num_layers",
+      "periphery_num_layers>0",
+      "Number of layers to subdivide the periphery boundary.");
+  params.addParam<std::string>("periphery_block_name", "Block name for periphery zone. ");
+  params.addParam<subdomain_id_type>(
+      "periphery_region_id", "ID for periphery zone for assignment of region_id extra element id.");
+  params.addRangeCheckedParam<Real>(
+      "desired_area",
+      0,
+      "desired_area>=0",
+      "Desired (maximum) triangle area, or 0 to skip uniform refinement");
+  params.addParam<std::string>(
+      "desired_area_func",
+      std::string(),
+      "Desired area as a function of x,y; omit to skip non-uniform refinement");
+  params.addParamNamesToGroup("periphery_block_name periphery_region_id outer_circle_radius "
+                              "mesh_periphery periphery_generator",
+                              "Periphery Meshing");
+  params.addParamNamesToGroup("outer_circle_num_segments desired_area desired_area_func",
+                              "PTMG-specific Meshing");
+  params.addParamNamesToGroup("periphery_num_layers", "PRMG-specific Meshing");
```

Comment:
this will look on the final page. I would leave it inside the other group OR
change the name to Periphery Meshing: PTMG specific (Parameters is added auatomatically)